Sources/Build/BuildPlan.swift

```diff
@@ -317,6 +318,20 @@ public final class SwiftTargetDescription {
         return args
     }
 
+    /// A list of compilation conditions to enable for conditional compilation expressions.
+    private var activeCompilationConditions: [String] {
+        var compilationConditions = ["-DSWIFT_PACKAGE"]
+
+        switch buildParameters.configuration {
+        case .debug:
+            compilationConditions += ["-DDEBUG"]
```

Comment:
Hm, I wonder if we should we also add this define to C language targets. /cc @ddunbar

Reply:
Tha makes sense. Should go to the `GCC_PREPROCESSOR_DEFINITIONS` (that's where the Xcode put it for newly created projects)

Reply:
We should add `-DDEBUG=1` here as well: https://github.com/apple/swift-package-manager/blob/master/Sources/Build/BuildPlan.swift#L198

Reply:
along with the `-DSWIFT_PACKAGE=1` or... is it set already?